gateway/ln-gateway/src/lnd.rs

```diff
@@ -309,10 +312,15 @@ impl ILnRpcClient for GatewayLndClient {
             .map_err(|status| LightningRpcError::FailedToGetRouteHints {
                 failure_reason: format!("Failed to list channels {status:?}"),
             })?
-            .into_inner();
+            .into_inner()
+            .channels;
+
+        // Take the channels with the largest incoming capacity
+        channels.sort_by(|a, b| b.remote_balance.cmp(&a.remote_balance));
+        channels.truncate(num_route_hints);
```

Comment:
Some redundancy in logic here, but probably less complexity than communicating channel balances to `gatewayd`.